server/src/main/java/org/opensearch/index/translog/transfer/BlobStoreTransferService.java

```diff
@@ -195,6 +230,40 @@ private void uploadBlob(
 
     }
 
+    private void asyncBlobUpload(
+        String fileName,
+        String remoteFileName,
+        long contentLength,
+        BlobPath blobPath,
+        WritePriority writePriority,
+        RemoteTransferContainer.OffsetRangeInputStreamSupplier inputStreamSupplier,
+        long expectedChecksum,
+        Map<String, String> metadata,
+        ActionListener<Void> completionListener
+    ) throws IOException {
+        BlobContainer blobContainer = blobStore.blobContainer(blobPath);
+        assert blobContainer instanceof AsyncMultiStreamBlobContainer;
```

Comment:
Can the assert be moved during the blobStore assignment (as a property of blobstore) instead of checking for every upload?